yellowbrick/features/jointplot.py

```diff
@@ -14,308 +14,369 @@
+
+    kind : str in {'scatter', 'hex'}, default: 'scatter'
+        The type of plot to render in the joint axes. Note that when kind='hex' the
+        target cannot be plotted by color.
+
+    hist : {True, False, None, 'density', 'frequency'}, default: True
+        Draw histograms showing the distribution of the variables plotted jointly.
+        If set to 'density', the probability density function will be plotted.
+        If set to True or 'frequency' then the frequency will be plotted.
+        Requires Matplotlib >= 2.0.2.
+
+    alpha : float, default: 0.65
+        Specify a transparency where 1 is completely opaque and 0 is completely
+        transparent. This property makes densely clustered points more visible.
+
+    {joint, hist}_kws : dict, default: None
+        Additional keyword arguments for the plot components.
 
     kwargs : dict
         Keyword arguments that are passed to the base class and may influence
         the visualization as defined in other Visualizers.
 
+    Attributes
+    ----------
+    corr_ : float
+        The correlation or relationship of the data in the joint plot, specified by the
+        correlation algorithm.
+
     Examples
     --------
 
-    >>> visualizer = JointPlotVisualizer()
-    >>> visualizer.fit(X,y)
-    >>> visualizer.poof()
-
-    Notes
-    -----
-    These parameters can be influenced later on in the visualization
-    process, but can and should be set as early as possible.
+    >>> viz = JointPlot(columns=["temp", "humidity"])
+    >>> viz.fit(X, y)
+    >>> viz.poof()
     """
 
-    def __init__(self, ax=None, feature=None, target=None,
-                 joint_plot='scatter', joint_args=None,
-                 xy_plot='hist', xy_args=None,
-                 ratio=5, space=.2, **kwargs):
-
-        # Check matplotlib version - needs to be version 2.0.0 or greater.
-        mpl_vers_maj = int(mpl.__version__.split(".")[0])
-        if mpl_vers_maj < 2:
-            warnings.warn((
-                "{} requires matplotlib major version 2 or greater. "
-                "Please upgrade."
-            ).format(self.__class__.__name__))
-
-        super(JointPlotVisualizer, self).__init__(ax, **kwargs)
-
-        self.feature = feature
-        self.target = target
-        self.joint_plot = joint_plot
-        self.joint_args = joint_args
-        self.xy_plot = xy_plot
-        self.xy_args = xy_args
-        self.ratio = ratio
-        self.space = space
-
-    def fit(self, X, y, **kwargs):
-        """
-        Sets up the X and y variables for the jointplot
-        and checks to ensure that X and y are of the
-        correct data type
-
-        Fit calls draw
+    # TODO: should we couple more closely with Rank2D?
+    correlation_methods = {
+        'pearson': lambda x, y: pearsonr(x,y)[0],
+        'spearman': lambda x, y: 1.0,
+        'covariance': lambda x, y: 1.0,
+        'kendalltau': lambda x, y: 1.0,
+    }
```

Comment:
Maybe we should more directly couple this with Rank2D?

Reply:
I don't think this is strictly necessary, though it would be useful to get someone to do a blog post showing the information conveyed via different correlation methods as depicted using YB visualizers. Potentially it might also be useful to add a note to #68, so that in case someone contributes new correlation measures we remember that we can incorporate them here, too.

Reply:
I like the idea of a blog post! I've added the note to #68. It could probably also use a line number for where `JointPlot` does correlations, but I can't get the line number until this is merged.